src/checkpoint/container/raw_ptr_serialize.h

```diff
@@ -81,29 +84,23 @@ void serializeRawPtr(SerializerT& s, T* ptr) {
 }
 
 /**
- * \brief Serialize FILE pointer \c ptr
- *
- * Only footprinting mode is supported at the moment. Counts the pointer size
- * without following it.
- *
- * \param[in] serializer serializer to use
- * \param[in] ptr pointer to serialize
+ * Note: do not follow void pointer.
  */
-template <typename SerializerT>
-void serialize(SerializerT& s, FILE* ptr) {
-  serializeFilePtr(s, ptr);
+template <
+  typename SerializerT
+>
+void serializeRawPtr(SerializerT& s, void* ptr) {
+  s.countBytes(ptr);
 }
 
+/**
+ * Note: do not follow FILE pointer as it might be an incomplete type on some
+ * platforms.
+ */
 template <
-  typename SerializerT,
-  typename = std::enable_if_t<
-    std::is_same<
-      SerializerT,
-      checkpoint::Footprinter
-    >::value
-  >
+  typename SerializerT
 >
-void serializeFilePtr(SerializerT& s, FILE* ptr) {
+void serializeRawPtr(SerializerT& s, FILE* ptr) {
```

Comment:
In the future, if we want to be very exacting about this, or need to support a broader range of possibly-incomplete types, we could use the `type_if_defined` template described [here](https://devblogs.microsoft.com/oldnewthing/20190712-00/?p=102690)